File: triptracking-ms/triptracking-lib/src/main/java/com/flashex/triptrackingmicroservice/lib/model/TripLog.java
package com.flashex.triptrackingmicroservice.lib.model;

import com.datastax.driver.core.DataType;
import lombok.*;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.CassandraType;
import org.springframework.data.cassandra.core.mapping.PrimaryKey;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;
import java.util.Date;
import java.util.List;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Table("tripslog")
public class TripLog {

    @PrimaryKey
    @PrimaryKeyColumn(name = "tripItineraryId",  ordinal = 0, type = PrimaryKeyType.PARTITIONED)
    @CassandraType(type = DataType.Name.TEXT)
    private String tripItineraryId;

    @CassandraType(type = DataType.Name.TIMESTAMP)
    private Date tripStart;

    @CassandraType(type = DataType.Name.TIMESTAMP)
    private Date tripEnd;

    @CassandraType(type = DataType.Name.LIST, typeArguments = DataType.Name.UDT, userTypeName = "packetlog")
    private List<PacketLog> packetLogs;

    @CassandraType(type = DataType.Name.TIMESTAMP)
    private Date plannedStartTime;

    @CassandraType(type = DataType.Name.TIMESTAMP)
    private Date plannedEndTime;
    @CassandraType(type = DataType.Name.UDT, userTypeName = "deliveryAddress")
    private DeliveryAddress originAddress;

}

File: common-ms/flashex-webapp/flashex-webapp/src/app/delivery-executive/interfaces/triplog.ts
export interface TripLog {
  tripItineraryId: string;
  tripStart: Date;
  tripEnd: Date;
  packetLogs: {
    packetId: string;
    deliveryAddress: {
      addressLine1: string;
      city: string;
      state: string;
      latitude: number,
      longitude: number,
      pincode: number
    };
    deliveryDescription: string;
    packetStatus: string;
  };
  plannedStartTime: Date;
  plannedEndTime: Date;
  originAddress: string;
}
export class TripLog {
  tripItineraryId: string;
  tripStart: Date;
  tripEnd: Date;
  packetLogs: {
    packetId: string;
    deliveryAddress: {
      addressLine1: string;
      city: string;
      state: string;
      latitude: number,
      longitude: number,
      pincode: number
    };
    deliveryDescription: string;
    packetStatus: string;
  };
  plannedStartTime: Date;
  plannedEndTime: Date;
  originAddress: string;
}
